File: app/src/main/java/com/incendiary/android/di/module/ApplicationModule.java
package com.incendiary.android.di.module;

import android.app.Application;
import android.content.Context;
import dagger.Module;
import dagger.Provides;
import com.incendiary.android.di.ApplicationContext;

/**
 * Provide application-level dependencies.
 */
@Module public class ApplicationModule {
  protected final Application mApplication;

  public ApplicationModule(Application application) {
    mApplication = application;
  }

  @Provides Application provideApplication() {
    return mApplication;
  }

  @Provides @ApplicationContext Context provideContext() {
    return mApplication;
  }
}

File: app/src/main/java/com/esafirm/androidboilerplate/di/module/ApplicationModule.kt
package com.esafirm.androidboilerplate.di.module

import android.app.Application
import android.content.Context
import com.esafirm.androidboilerplate.di.ApplicationContext
import dagger.Module
import dagger.Provides

/**
 * Provide application-level dependencies.
 */
@Module class ApplicationModule(val mApplication: Application) {

	@Provides internal fun provideApplication(): Application {
		return mApplication
	}

	@Provides @ApplicationContext internal fun provideContext(): Context {
		return mApplication
	}
}
